### DISCUSSION TITLE
414 Request-URI Too Large

### DISCUSSION THREAD

**Discussion by scouech** [directus/directus#5419] [Archive]
Hi, 

When I go to a user role I get an error:
`"<html>\r\n<head><title>414 Request-URI Too Large</title></head>\r\n<body>\r\n<center><h1>414 Request-URI Too Large</h1></center>\r\n<hr><center>nginx/1.18.0 (Ubuntu)</center>\r\n</body>\r\n</html>\r\n"
`
Any idea? 

Cheers

**Comment by scouech** [comment]
I think I know why, I do have a very large number of users under one role.. is there a way to not display the user list on the user role settings page?

**Reply by tylerforesthauser** [reply]
To me, this seems more related to your nginx config but I could be wrong...

`large_client_header_buffers`

http://nginx.org/en/docs/http/ngx_http_core_module.html#large_client_header_buffers

**Reply by scouech** [reply]
yep I think you are right @tylerforesthauser, I did change the header buffers size and it works.. only thing I'm worried is that is it going to continue to give me the error as the number of users increases...

**Comment by rijkvanzanten** [comment]
Pretty sure the problem is the fact that the filter for the users gets too large for the URL to handle (eg `/users?filter[id][_in]=13114266-e5da-4c43-a918-232692d95559,13114266-e5da-4c43-a918-232692d95559,13114266-e5da-4c43-a918-232692d95559,13114266-e5da-4c43-a918-232692d95559,13114266-e5da-4c43-a918-232692d95559` etc).

We just added support for doing a `SEARCH` request that can prevent this from being a problem, so that sounds like the ideal solution here 👍🏻 (https://docs.directus.io/reference/api/introduction/#search-http-method)

**Reply by scouech** [reply]
Thanks! Is this going to be how the admin side does it too then? Because right now the admin panel gives me the error when accessing the user role (as the users are listed below the permissions on the role page)

**Reply by rijkvanzanten** [reply]
Yeah!

**Reply by scouech** [reply]
amazing! 🤩  
You might be aware of it, and it's probably the same problem, but on the Roles & Permissions page where it shows all the roles, with a column with the number of users, it stops counting after 100. So no matter how many user I have pass 100 it always shows 100 users

**Reply by rijkvanzanten** [reply]
Yeah that's https://github.com/directus/directus/issues/3743. We can't automatically fetch the full count, as that would be way to heavy, so it's accurate up until 100, and then shows 100+

**Reply by benhaynes** [reply]
I know the pagination is handled in #3743, but do we want to add a ticket to update the Role to use `SEARCH`?

**Reply by rijkvanzanten** [reply]
This is o2m/m2m interfaces in general, not just role 👍🏻

**Reply by benhaynes** [reply]
Ahh, gotcha — and do we have a dedicated issue for tracking that?

**Reply by rijkvanzanten** [reply]
@benhaynes https://github.com/directus/directus/issues/5540 👍🏻
